### DISCUSSION TITLE
@CasePathable fails to compile when nested in private type

### DISCUSSION THREAD

**Discussion by pietrasm** [pointfreeco/swift-case-paths#208] [General]
**Describe the bug**
`@CasePathable` generates code that fails to compile when the enum is nested in a `private` type.

**To Reproduce**
```swift
struct A {
    private struct B {
        @CasePathable
        enum C {}
        // Expanded Macro
        extension A.B.C: CasePaths.CasePathable, CasePaths.CasePathIterable { // 🛑 Error: 'B' is inaccessible due to 'private' protection level
        }
    }
}
```

**Expected behavior**
Successful compilation.

**Environment**
  - swift-case-paths 1.5.6
  - Xcode 16.0
  - Swift 6.0
  - OS: iOS 18.0.1

**Additional context**
Relaxing `private` to `fileprivate` can be used as a workaround.

**Comment by mbrandonw** [comment]
Hi @pietrasm, unfortunately I'm not sure there is anything we can do to fix this. Our macro is merely expanding an "extension macro" to add some conformances to the `C` type, and Swift is interpreting that to mean `extension A.B.C`. I think for now using `fileprivate` is a good workaround, or you can add the conformances yourself and then the macro will skip them:

```swift
@CasePathable
enum C: CasePathable, CasePathIterable {}
```

Since this isn't an issue with the library I am going to convert it to a discussion. Please feel free to continue the conversation over there! And if you have the time I recommend filing an issue with Apple because it does seem like something that should be allowed.

**Reply by pietrasm** [reply]
Hi @mbrandonw, I suspected this might be a swift issue. I have just reported it: https://github.com/swiftlang/swift/issues/77023

Also, thanks for pointing out that manually conforming the type could be used as a workaround as well.
